system/security/plugins/testauthSecurity/testauthSecurity.cpp

```diff
@@ -0,0 +1,591 @@
+            feature.getProp("@description", description);
+            ISecResourceList* rlist = authmap->queryResourceList(pathStr);
+            if(rlist == nullptr)
+            {
+                rlist = createResourceList(pathStr, secureContext);
+                authmap->add(pathStr, rlist);
+            }
+            if (!rstr.isEmpty())
+            {
+                ISecResource* rs = rlist->addResource(rstr);
+                SecAccessFlags requiredaccess = str2perm(required);
+                rs->setRequiredAccessFlags(requiredaccess);
+                rs->setDescription(description);
+            }
+        }
+        return authmap.getClear();
+    }
+
+    IAuthMap* createSettingMap(IPropertyTree* authConfig, IEspSecureContext* secureContext = nullptr) override
+    {
+        return nullptr;
+    }
+
+    SecAccessFlags getAccessFlag(SecResourceType rtype, ISecUser& user, const char* resourceName, IEspSecureContext* secureContext)
+    {
+        SecAccessFlags  resultFlag = SecAccess_Unavailable;
+        try
+        {
+            if (!authenticate(user))
+                return resultFlag;
+
+            StringBuffer userN;
+            userN.append(user.getName());
+            CUserAccessPtr* userAccess = userAccessMap.getValue(userN);
+            if (!userAccess)
+                return resultFlag;
+
+            CUserAccessPtr userAccessPtr = *userAccess;
+
+            if(rtype == RT_FILE_SCOPE)
+                resultFlag = userAccessPtr->queryFileScopeAccess(resourceName);
+            else if(rtype == RT_WORKUNIT_SCOPE)
+                resultFlag = userAccessPtr->queryECLWUScopeAccess(resourceName);
+            else
+                resultFlag = userAccessPtr->queryFeatureAccess(resourceName);
+        }
+        catch (IException* e)
+        {
+            EXCLOG(e, "CTestAuthSecurityManager::getAccessFlag(...) exception caught");
+            e->Release();
+        }
+        catch (...)
+        {
+            // Need better logging
+            DBGLOG("CTestAuthSecurityManager::getAccessFlag(...) exception caught");
+        }
+
+        return resultFlag;
+    }
+
+protected:
+
+    //ISecManager
+    bool authenticate(ISecUser& user)
+    {
+        if (user.credentials().getSessionToken() != 0 || user.getAuthenticateStatus() == AS_AUTHENTICATED)
+        { //Authenticated.
+            return true;
+        }
+
+        StringBuffer userN, password;
+        userN.set(user.getName());
+        if (userN.isEmpty())
+            return false;
+
+        CUserAccessPtr* userAccess = userAccessMap.getValue(userN);
+        if (!userAccess)
+            return false;
+
+        CUserAccessPtr userAccessPtr = *userAccess;
+        password.set(user.credentials().getPassword());
+        return streq(password.str(), userAccessPtr->queryPassword());
+    }
+
+    bool authorize(ISecUser& user, ISecResourceList* Resources, IEspSecureContext* secureContext = nullptr) override
```

Comment:
minor: I think this is public in the base interface, so not sure it makes sense for it to be protected (and other functions)

Reply:
Will change to public.